build/build/src/enso.rs

```diff
@@ -100,6 +100,9 @@ impl BuiltEnso {
             // This flag enables assertions in the JVM. Some of our stdlib tests had in the past
             // failed on Graal/Truffle assertions, so we want to have them triggered.
             .set_env(JAVA_OPTS, &ide_ci::programs::java::Option::EnableAssertions.as_ref())?;
+        if test_path.as_str().contains("_Internal_") {
+            command.arg("--disable-private-check");
+        }
```

Comment:
I like this approach :)

Ideally it would be nice if this was configurable through `package.yaml` (totally not a thing in scope of this PR obviously). But in the current context this check seems like a good heuristic.